docs/options.md

```diff
@@ -327,6 +327,32 @@ Platform-specific variants also available:<br/>
 CIBW_TEST_EXTRAS: test,qt
 ```
 
+### `CIBW_BEFORE_TEST` {: #before-test}
+> Execute a shell command preparing test environment
+
+A shell command to run after setup test environment. This option allows you to run a command in **each** Python environment before the `pip wheel` command. This is useful if you need to install non pip package, change values of environment variables
+or perform multi step pip installation (like install `scikit-build` or `cython` before install test package)
+
+If dependencies are required to build your wheel (for example if you include a header from a Python module), set this to `pip install .`, and the dependencies will be installed automatically by pip. However, this means your package will be built twice - if your package takes a long time to build, you might wish to manually list the dependencies here instead.
+
+The active Python binary can be accessed using `python`, and pip with `pip`; `cibuildwheel` makes sure the right version of Python and pip will be executed. `{project}` can be used as a placeholder for the absolute path to the project's root and will be replaced by `cibuildwheel`.
+
+On Linux and macOS, the command is run in a shell, so you can write things like `cmd1 && cmd2`.
+
+Platform-specific variants also available:<br/>
+ `CIBW_BEFORE_TEST_MACOS` | `CIBW_BEFORE_TEST_WINDOWS` | `CIBW_BEFORE_TEST_LINUX`
+
+#### Examples
+```yaml
+# install packages needed to build test dependencies
+CIBW_BEFORE_TEST: pip install cmake scikit-build
+
+# install test dependencies with overwritten environment variables. 
+CIBW_BEFORE_TEST: CC=gcc CXX=g++ pip install -r requirements.txt
```

Comment:
Curious - why would this be necessary?

Reply:
My case is that i want to build package with OMP support for MacOS. MacOS clang does not support this. So I need to use brew g++ for this. But one of test dependencies need to be compiled with system `clang`. Otherwise I got such message:

```
python(53258,0x7fff9481d380) malloc: *** error for object 0x10f526c00: pointer being freed was not allocated
*** set a breakpoint in malloc_error_break to debug
```

Other case which I can imagine is that my package is build against newest version of some library. And test dependency is build against older (system one). And I need to change `LD_LIBRARY_PATH` etc to avoid collision.

Reply:
This confusion makes me wonder; wouldn't a different example be better for the docs, then?

Reply:
What do you think about 
`CFLAGS="-I ... -L ..." pip install package name` 
for custom library location?

Reply:
Huh, maybe. But .. that's again something where you'd ask the question where a user would ask why this is not `CIBW_TEST_ENVIRONMENT`...

Reply:
Totally different example. Install non pip package like: https://github.com/LLNL/zfp

For this...  (original example)
MacOS system clang does not support OpenMP. So if package needs OpenMP support then there is need to compile it with gcc/g++ (ex from brew). 

For second. I meet problem when test for one package fail on Azure Pipeline but not fail on my computer. And this fail was connected with package used in test and build from source. And for me possibility to control compilation of single package is big profit.